### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by RedGuy12**

### CODE REVIEW

**Comment by YandeMC** [comment]
one day... (not pressuring take your time)

**Comment by RedGuy12** [comment]
image

**Comment by YandeMC** [comment]
> image
> 

i saw

File: modules/auto/scratch.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+	for (const match of matches) {
+		if (match.startsWith("<") && match.endsWith(">")) continue;
+
+		const start = match.startsWith("http") ? 0 : 1,
+			end = match.length - (/[\w!#$&'()*+,./:;=?@~-]$/.test(match) ? 0 : 1);
+		const urlParts =
+			match
+				.slice(start, end)
+				.replace(
+					/https:\/\/scratch.mit.edu\/discuss\/topic\/\d+(?:\?page=\d+)?#post-/,
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply modules/auto/scratch.ts:35]: oh

**Comment by RedGuy12** [comment]
note to future self: use "current" rev of all commits up to here when resolving conflicts

**Comment by RedGuy12** [comment]
not done yet, but ill finish on main branch

**Pull request merged**